examples/hello_world/finance/demo.py

```diff
@@ -0,0 +1,112 @@
+import time
+import random
+from collections import OrderedDict
+from maro.simulator import Env, DecisionMode
+from maro.simulator.scenarios.finance.common.common import MarketOrder, LimitOrder, StopOrder, StopLimitOrder, OrderMode, ActionType, DecisionEvent, CancelOrder, OrderDirection
+from maro.simulator.utils.common import tick_to_frame_index
+
+auto_event_mode = False
+start_tick = 10226  # 2019-01-01
+durations = 100
+max_ep = 1
+snapshot_resolution = 1
+
+env = Env(scenario="finance", topology="demo", start_tick=start_tick, durations=durations, decision_mode=DecisionMode.Joint, snapshot_resolution=snapshot_resolution)
+
+print(env.summary)
+for ep in range(max_ep):
+    metrics = None
+    decision_evts = None
+    is_done = False
+    actions = None
+
+    while not is_done:
+        metrics, decision_evts, is_done = env.step(actions)
+
+        ep_start = time.time()
+        actions = []
+        if not is_done:
+            for decision_event in decision_evts:
+                if decision_event.action_type == ActionType.order:
+                    # stock trading decision
+                    stock_index = decision_event.item
+                    action_scope = decision_event.action_scope
+                    last_frame_idx = tick_to_frame_index(start_tick, env.tick-1, snapshot_resolution)
+                    min_amount = action_scope.buy_min
+                    max_amount = action_scope.buy_max
+                    sell_min_amount = action_scope.sell_min
+                    sell_max_amount = action_scope.sell_max
+                    supported_order_types = action_scope.supported_order
+                    print("<<<<    trade decision_event", stock_index, min_amount, max_amount, sell_min_amount, sell_max_amount)
+                    # qurey snapshot for stock information
+                    cur_env_snap = env.snapshot_list['stocks']
+                    holding = cur_env_snap[last_frame_idx:int(decision_event.item):"account_hold_num"][-1]
+                    cost = cur_env_snap[last_frame_idx:int(decision_event.item):"average_cost"][-1]
+                    opening_price = cur_env_snap[last_frame_idx:int(decision_event.item):"opening_price"][-1]
+                    closing_price = cur_env_snap[last_frame_idx:int(decision_event.item):"closing_price"][-1]
+                    highest_price = cur_env_snap[last_frame_idx:int(decision_event.item):"highest_price"][-1]
```

Comment:
why need int cast for decision_event.item?

Reply:
use can extract them in a single query

Reply:
removed

Reply:
fix the float value of item